pkg/policyfilter/map.go

```diff
@@ -0,0 +1,223 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package policyfilter
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"unsafe"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/tetragon/pkg/bpf"
+	"github.com/cilium/tetragon/pkg/kernels"
+	"github.com/cilium/tetragon/pkg/option"
+)
+
+const (
+	MapName = "policy_filter_maps"
+)
+
+// map operations used by policyfilter.
+
+// PfMap is a simple wrapper for ebpf.Map so that we can write methods for it
+type PfMap struct {
+	*ebpf.Map
+}
+
+// newMap returns a new policy filter map.
+func newPfMap() (PfMap, error) {
+	// use the generic kprobe program, to find the policy filter map spec
+	objName, _ := kernels.GenericKprobeObjs()
+	objPath := path.Join(option.Config.HubbleLib, objName)
+	spec, err := ebpf.LoadCollectionSpec(objPath)
+	if err != nil {
+		return PfMap{}, fmt.Errorf("loading spec for %s failed: %w", objPath, err)
+	}
+	policyMapSpec, ok := spec.Maps[MapName]
+	if !ok {
+		return PfMap{}, fmt.Errorf("%s not found in %s", MapName, objPath)
+	}
+
+	// bpf-side sets max_entries to 1. Later kernels (5.10) can deal with
+	// inserting a different size of inner-map, but for older kernels, we
+	// fix the spec here.
+	policyMapSpec.InnerMap.MaxEntries = polMapSize
+
+	ret, err := ebpf.NewMap(policyMapSpec)
+	if err != nil {
+		return PfMap{}, err
+	}
+
+	mapDir := bpf.MapPrefixPath()
+	pinPath := filepath.Join(mapDir, MapName)
+	os.Remove(pinPath)
+	os.Mkdir(mapDir, os.ModeDir)
```

Comment:
Here you just remove the map. Why do you need to call `mkdir` again?

Reply:
There are cases where the map does not exist (os.Remove() will just return an error, which we do not check) and neither does the directory. We could check if the directory exists or not but not sure if it's worth it.

Reply:
Sounds reasonable, thanks!